zerver/tests/test_events.py

```diff
@@ -997,6 +998,45 @@ def test_custom_profile_fields_events(self) -> None:
         error = schema_checker('events[0]', events[0])
         self.assert_on_error(error)
 
+    def test_custom_profile_field_data_events(self) -> None:
+        schema_checker = self.check_events_dict([
+            ('type', equals('realm_user')),
+            ('op', equals('update')),
+            ('person', check_dict_only([
+                ('user_id', check_int),
+                ('custom_profile_field', check_dict_only([
+                    ('id', check_int),
+                    ('value', check_none_or(check_string)),
+                ])),
+            ])),
+        ])
+
+        realm = get_realm("zulip")
+        field_id = realm.customprofilefield_set.get(realm=realm, name='Biography').id
+        field = {
+            "id": field_id,
+            "value": "New value",
+        }
+        events = self.do_test(
+            lambda: notify_user_update_custom_profile_data(
```

Comment:
This is the wrong way to construct such a test; the whole model for these tests is that we're doing the actual state change and confirming we handle the race correctly.  Read http://zulip.readthedocs.io/en/latest/subsystems/events-system.html to understand this test class fully.

So don't call the `notify` function, call the `do_update_user_custom_profile_data` function (and don't set `state_change_expected=False`; that's only for very rare special cases like typing notifications that are not reflected in the database).